### DISCUSSION TITLE
Add Enum Support for GraphQL

### DISCUSSION THREAD

**Discussion by HarunKilic** [directus/directus#7985] [Feature Requests]
Hi guys, 

It would be really nice to have enum support in graphql types. 
As enum, i mean eg. `dropdown`. In graphql it returns just as `String` type. Would be nice if it returned as the values provided to the dropdown instead.

**Comment by rijkvanzanten** [comment]
Sort of duplicate of https://github.com/directus/directus/discussions/4887

**Comment by rijkvanzanten** [comment]
We can't just expose the "choices" that are configured on a dropdown interface, as those are "flexible". By that I mean you can have a value that's not in the enum list (through "Allow Other"), and you can have those choices be changed / affected by the values of other fields on the form (through conditional fields). In order to expose enum values in GraphQL, we'll have to support "actual" ENUMs (which don't have this flexibility) first

**Reply by HarunKilic** [reply]
I see the challenge. Is it added to the milestone or will it be in the backlog for a while?

**Reply by rijkvanzanten** [reply]
Adding native ENUM type support is not currently a in-works item

**Reply by HarunKilic** [reply]
Okay thanks :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
